### ISSUE TITLE
ReactiveCassandraTable breaks on larger tables / how to stream results of a large query

### ISSUE THREAD

**Issue by ksilin** [outworkers/phantom#652]
My use case - copying and transforming entire tables with millions of entries programmatically. 

* Expected behaviour: 

`table.publisher()` handles batching for large result sets. 

same for manually constructed large queries, e.g.

`Streams.enumeratorToPublisher(table.select.all().fetchEnumerator())`

* actual behavior: timeout errors: 

` Failed to execute query SELECT * FROM keyspace.table;
com.datastax.driver.core.exceptions.OperationTimedOutException: [/<IP>:9042] Timed out waiting for server response`

The result set is far too big, sure. But this is what I hoped a stream could help me with by handling pagination. 

* how to reproduce: 

use the standard streaming example with a large table - 10⁶ records should be enough. 

https://github.com/thiagoandrade6/cassandra-phantom/blob/master/src/test/scala/com/cassandra/phantom/modeling/test/stream/SongsStreaming.scala

EDIT: adding system specs

* JDK 1.8.0_112.15, managed by DSE OpsCenter
* Phantom 2.6.1
* C* 3.0.11 over DSE 5.0.6
* Cluster setup with 2 DCs, 3 nodes each.
* DC1 - master on AWS, DC2 - replicas on rented dedicated servers
* AWS nodes have 4 cores and 30 GB RAM each
* OpsCenter manages the heap, with max-heap set to 'auto'
* read request timeout of 5000, range and request timeouts of 10000, streaming socket timeout 86400000

**Comment by alexflav23** [comment]
Hi @ksilin,

Ok, can you help me out with some more things, like JDK version, phantom version, Cassandra version, ram, configuration you used for the JDK and C*? I'd like to be able to reproduce properly, and this is not enough information for me to be helpful just yet.

Regards.

**Comment by ksilin** [comment]
Thank you for looking into it @alexflav23. Have added the specs. Please let me know if there is more info I can provide. 

I will try to reproduce on a local single-node cluster with default settings, so you won't need to spin up an entire DC :)

**Comment by alexflav23** [comment]
Hi @ksilin Let's give it another try on 2.8.0, you should see huge performance improvements in throughput. I'll update the necessary things in the example you linked to with a PR shortly.

**Comment by alexflav23** [comment]
Hi @ksilin Even more performance improvements of huge significance on 2.9.0, we have removed all runtime logic for low level Byte I/O, which should give a huge boost to anything of decent size. Any time/luck trying it out?

**Comment by ksilin** [comment]
Hi @alexflav23, have tried 2.8.1 and can confirm, it works beautifully. Thank you! Will try 2.9 soon, the API changes were in the way of a drop-in replacement. 

Do you happen to have an ETA for a 2.12 release of phantom-streams? Our projects are all on 2.12 now, so currently I have to use [alpakka](https://github.com/akka/alpakka) to stream out query results.

**Comment by alexflav23** [comment]
H @ksilin, stand by for #693, it should go in today/tomorrow, the build is already passing, I'm just working to re-enable some more sizeable performance tests.

**Comment by alexflav23** [comment]
Hi @ksilin Scala 2.12 support now available in 2.9.2. I'm going to close this PR as everything is hopefully fixed!

**Comment by ksilin** [comment]
Hi @alexflav23, have tried 2.9.1 and 2.9.2. Excited about the perf gains and 2.12 compat. You guys rock! 

A micro migration guide for table classes:

```
class MyModel extends CassandraTable[ConcreteMyModel, MyRecord] {

  object recordId   extends StringColumn(this) with PartitionKey
  object createdAt  extends OffsetDateTimeColumn(this)
  object languages  extends ListColumn[String](this)
```

turns to

```
abstract class MyModel extends Table[ConcreteMyModel, MyRecord] {

  object recordId   extends StringColumn with PartitionKey
  object createdAt  extends OffsetDateTimeColumn(this)
  object languages  extends ListColumn[String]
```

the model class does not have to be abstract if you provide `keyspace` & `session`

java 8 time classes (from `phantom-jdk8`) still need to be parametrized by the table.
